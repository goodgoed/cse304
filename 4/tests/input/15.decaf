class Simple {
    void test(int a){
        a = +a;
        a = -a;
    }
}