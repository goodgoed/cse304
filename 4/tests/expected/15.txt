--------------------------------------------------------------------------
Class Name: Simple
Superclass Name: 
Fields:
Constructors:
Methods:
METHOD: 1, test, Simple, private, instance, void
Method Parameters: 1
Variable Table:
VARIABLE: 1, a, formal, int
Method Body:
Block([
Expr(Assign(Variable(1), Variable(1), int, int)),
Expr(Assign(Variable(1), Unary(uminus, Variable(1)), int, int))
])
--------------------------------------------------------------------------
